src/braket/circuits/diagram_builders/braket_circuit_drawer.py

```diff
@@ -0,0 +1,402 @@
+            circle1 = Circle((x_ctr, y1), r, color="black")
+            circle2 = Circle((x_ctr, y2), r, color="black")
+            ax.add_patch(circle1)
+            ax.add_patch(circle2)
+            # Target circle with “+”
+            circle_t = Circle((x_ctr, yt), r, fill=False, edgecolor="black", linewidth=1.0)
+            ax.add_patch(circle_t)
+            ax.text(x_ctr, yt, "+", ha="center", va="center", fontsize=8)
+
+            # Connect vertical line spanning all three
+            ymin = min(y1, y2, yt) + r
+            ymax = max(y1, y2, yt) - r
+            ax.vlines(x_ctr, ymin, ymax, color="black", linewidth=1.0)
+
+        # 5) Measurement(s) (MEASURE) on one or more qubits
+        elif gate.startswith("MEASURE"):
+            for q in qubits:
+                yq = y_of(q)
+                _draw_meter(x_ctr, yq)
+
+        # 6) Generic multi-qubit gates (e.g., SWAP, custom gates, etc.)
+        else:
+            # Determine top and bottom y-coordinates among all involved qubits
+            y_vals = [y_of(q) for q in qubits]
+            y_top = max(y_vals) + 0.3 * dy
+            y_bot = min(y_vals) - 0.3 * dy
+            width_gate = 0.8 * dx
+            height_gate = y_top - y_bot
+
+            rect = Rectangle(
+                (x_ctr - width_gate / 2, y_bot),
+                width_gate, height_gate,
+                facecolor="white", edgecolor="black", linewidth=1.0,
+            )
+            ax.add_patch(rect)
+            y_center = (y_top + y_bot) / 2
+            ax.text(x_ctr, y_center, gate, ha="center", va="center", fontsize=8)
+
+    ax.set_xlim(x_min, x_max + dx)
+    ax.set_ylim(-n_qubits * dy - dy / 2, dy / 2)
+    plt.tight_layout()
+    return fig, ax
+
+def generate_circuit_latex(circuit, initial_states=None):
+    """
+    Generate LaTeX (quantikz) code for the given Braket Circuit.
+
+    Parameters:
+      - circuit: braket.circuits.Circuit object
+      - initial_states: optional list of initial state labels (e.g., ['|0>', '|+>', ...])
+        If provided, each will be placed in the \lstick{} for that qubit. Otherwise, defaults to \ket{0} for all.
+
+    Returns:
+      - latex_code: a string containing a `quantikz` environment ready to paste into a .tex file.
+    """
+    instructions, n_qubits = parse_circuit(circuit)
+    if n_qubits == 0:
+        raise ValueError("The circuit contains no qubits to generate LaTeX.")
+
+    n_cols = len(instructions)
+    # Initialize a matrix of "\qw" tokens: size n_qubits x n_cols
+    matrix = [["\\qw" for _ in range(n_cols)] for _ in range(n_qubits)]
+
+    for instr_dict in instructions:
+        gate = instr_dict["gate"]
+        qubits = instr_dict["qubits"]
+        col = instr_dict["col_index"]
+        instr = instr_dict["instr"]
+
+        if not qubits:
+            continue
+
+        # 1) Parameterized single-qubit gates (RX, RY, RZ, PHASE, U3, etc.)
+        if len(qubits) == 1 and gate in ["RX", "RY", "RZ", "PHASE", "U3"]:
```

Comment:
same comment as the matplotlib function - we shouldn't hard-code the list of gates